### PR TITLE
Added o-tracking to gmail signatures

### PR DESCRIPTION
**Pull request by seanmtracey**
We can track the person clicking the send button, but not hitting cmd + return to send. GMail's crazy element naming/AJAXy nature makes it really difficult to hook into the action of sending an email, with this PR we infer that an email has been sent by listening for a click on the send button, and checking for a success notification 3 seconds later.

### CODE REVIEW

File: src/scripts/main.js [hunk 1]
```diff
@@ -50,6 +55,25 @@ function populateSignatures(data, force) {
...
+		const submitButton = parent.querySelector(".gU.Up");
+		
+		submitButton.addEventListener('click', function(){
+			
+			// Circumstances exist where a user can click send, and the email doesn't send
+			//  - such as forgetting to put a recipient in the dialog. So, when the send
+			// button is clicked, we wait 3 seconds, and see if there's a success dialog i
+			// somewhere in the page.
+			 			
+			setTimeout(function(){
```
Comment (JakeChampion) [inline src/scripts/main.js:68]: Could/Should we replace this with a mutation observer on the success dialog?

File: src/scripts/main.js [hunk 2]
```diff
@@ -60,7 +84,7 @@ function populateSignatures(data, force) {
 		if(parent === null){
 			const containingElement = findParentElementByAttribute(message, 'class', 'iN');
 			const addAnywayApendee = containingElement.querySelector('.gU.OoRYyc:not([data-sig-pone-assigned="true"])');
-
+
```
Comment (JakeChampion) [inline src/scripts/main.js]: Extra whitespace

File: src/scripts/main.js (hunk 1 above)
Comment (seanmtracey) [inline src/scripts/main.js:68]: I'd prefer to trigger an event knowing that our user had clicked something and then use the DOM to check that what we think happened (sending an email) actually happened.

**Comment by JakeChampion** [comment]
Currently this will have false positives if a user adds an email signature and then manually removes it. To remove these false positives we can check to see if the email contains links which have http://ftsig, if the email contains this then we would create our timeout function to fire the tracking event.

**Pull request merged**